### PR TITLE
refactor(static_obstacle_avoidance): organize params for drivable lane

### PR DESCRIPTION
**Pull request by satoshi-ota**
## Description

We don't expect to set params for drivable lane as follow:

```yaml
use_adjacent_lane: false
use_opposite_lane: true
```

However, we were able to do that in previous config structure.

I fixed it to prevent unexpected paramter configuration.

## Related links

**Parent Issue:**

:warning: Please approve this PR at first.
- https://github.com/autowarefoundation/autoware_launch/pull/1042

## How was this PR tested?

Psim

```yaml
      # drivable lane setting. this module is able to use not only current lane but also right/left lane
      # if the current lane(=lanelt::Lanelet) and the rignt/left lane share the boundary(=lanelet::Linestring) in HDMap.
      # "current_lane"           : use only current lane. this module doesn't use adjacent lane to avoid object.
      # "same_direction_lane"    : this module uses same direction lane to avoid object if need.
      # "opposite_direction_lane": this module uses both same direction and opposite direction lane.
      use_lane_type: "same_direction_lane"
```
Screenshot from 2024-06-27 10-25-43

```yaml
      # drivable lane setting. this module is able to use not only current lane but also right/left lane
      # if the current lane(=lanelt::Lanelet) and the rignt/left lane share the boundary(=lanelet::Linestring) in HDMap.
      # "current_lane"           : use only current lane. this module doesn't use adjacent lane to avoid object.
      # "same_direction_lane"    : this module uses same direction lane to avoid object if need.
      # "opposite_direction_lane": this module uses both same direction and opposite direction lane.
      use_lane_type: "opposite_direction_lane"
```
Screenshot from 2024-06-27 10-22-50

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

File: planning/behavior_path_planner/autoware_behavior_path_static_obstacle_avoidance_module/schema/static_obstacle_avoidance.schema.json [hunk 1]
```diff
@@ -22,14 +22,10 @@
...
-          "default": "true"
-        },
-        "use_opposite_lane": {
-          "type": "boolean",
-          "description": "Extend avoidance trajectory to opposite direction lane. `use_adjacent_lane` must be `true` to take effects.",
+        "use_lane_type": {
+          "type": "string",
+          "enum": ["current_lane", "same_direction_lane", "opposite_direction_lane"],
+          "description": "Drivable lane configuration.",
           "default": "true"
```
Comment (go-sakayori) [inline planning/behavior_path_planner/autoware_behavior_path_static_obstacle_avoidance_module/schema/static_obstacle_avoidance.schema.json:29]:
```suggestion
          "default": "current_lane"
```

Comment (satoshi-ota) [reply planning/behavior_path_planner/autoware_behavior_path_static_obstacle_avoidance_module/schema/static_obstacle_avoidance.schema.json:29]: Ahhhhhhhh.....

Comment (satoshi-ota) [inline planning/behavior_path_planner/autoware_behavior_path_static_obstacle_avoidance_module/schema/static_obstacle_avoidance.schema.json:29]:
```suggestion
          "default": "opposite_direction_lane"
```

Comment (satoshi-ota) [reply planning/behavior_path_planner/autoware_behavior_path_static_obstacle_avoidance_module/schema/static_obstacle_avoidance.schema.json]: I fixed it in https://github.com/autowarefoundation/autoware.universe/pull/7715/commits/65ace5a8ca733bf6a020abf85fcf90c098520b7f.

**Review by go-sakayori — APPROVED** [review]

**Pull request merged**